test/gtest/uct/test_pending.cc

```diff
@@ -221,114 +300,58 @@ UCS_TEST_SKIP_COND_P(test_uct_pending, pending_op,
                      !check_caps(UCT_IFACE_FLAG_AM_SHORT |
                                  UCT_IFACE_FLAG_PENDING))
 {
-    uint64_t send_data = 0xdeadbeef;
-    ucs_status_t status;
-    unsigned i, iters, counter = 0;
+    uint64_t init_data = 0xdeadbeef;
+    uint64_t send_data = init_data;
+    unsigned counter   = 0;
 
     initialize();
 
-    iters = 1000000 / ucs::test_time_multiplier();
-
     /* set a callback for the uct to invoke for receiving the data */
-    install_handler_sync_or_async(m_e2->iface(), 0, am_handler, &counter);
+    install_handler_sync_or_async(m_e2->iface(), AM_ID, am_handler, &counter);
 
     /* send the data until the resources run out */
-    i = 0;
-    while (i < iters) {
-        status = uct_ep_am_short(m_e1->ep(0), 0, test_pending_hdr, &send_data,
-                                 sizeof(send_data));
-        if (status != UCS_OK) {
-            if (status == UCS_ERR_NO_RESOURCE) {
+    send_ams_and_add_pending(m_e1->ep(0), PENDING_HDR, &send_data, false);
 
-                pending_send_request_t *req = pending_alloc(send_data);
+    unsigned num_sends = send_data - init_data;
```

Comment:
mayb return num_sends from send_ams_and_add_pending(), rather than calculating it indirectly from send_data?